sdk/storage/azblob/CHANGELOG.md

```diff
@@ -22,7 +22,7 @@
 
 ### Other Changes
 
-* Avoid allocations when storing/fetching a slice via sync.Pool.
+* The `UploadStream()` methods now use memory mapped files for buffers in order to reduce heap allocations/fragmentation.
```

Comment:
Few concerns here with memory mapping:
1. At process level we also have limitations on how many files we can memory map at a given time. If user application does too many upload/downloads in parallel that will be an issue here.
2. Memory mapping a file will increase our memory usage and we may put system under resource crunch with memory mapping huge files for upload/download.

Reply:
A memory mapping doesn't require a file descriptor, so these don't count towards the process' quota. Also, these are anonymous memory mappings so no for descriptor is even used at all. Also, we're not doing any file uploads or downloads here. We're just creating the anonymous memory mappings for this upload stream method. The number and size of the memory maps buffers is determined by the customer options so they are controlled directly by the application developer.

Reply:
I've updated the text here to clarify that this uses anonymous memory mapped files.

Reply:
Also, regarding point 2 above, the total memory allocated is identical compared to using byte slices (sans any differences in overhead imposed by the OS).  The difference is that the anonymous memory mapped files aren't part of the GC heap, so using them reduces GC pressure/fragmentation.